unison-core/src/Unison/Project.hs

```diff
@@ -386,8 +387,8 @@ instance TryFrom Text (ProjectAndBranch ProjectName (Maybe ProjectBranchName)) w
 --   3. project/branch
 projectWithOptionalBranchParser :: Megaparsec.Parsec Void Text (ProjectAndBranch ProjectName (Maybe ProjectBranchName))
 projectWithOptionalBranchParser = do
-  (project, hasTrailingForwardSlash) <- projectNameParser
-  branch <- if hasTrailingForwardSlash then optional projectBranchNameParser else pure Nothing
+  (project, hasTrailingSlash) <- projectNameParser
+  branch <- if hasTrailingSlash then optional (projectBranchNameParser False) else pure Nothing
```

Comment:
Maybe my suggestion would be weird here if you have to convert between HasTrailingSlash and AllowTrailingSlash